File: src/main/java/il/tutorials/similaritysearch/java/EmbeddingsController.java
package il.tutorials.similaritysearch.java;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class EmbeddingsController {


    private final EmbeddingsService embeddings;

    public EmbeddingsController(EmbeddingsService embeddings) {
        this.embeddings = embeddings;
    }

    @GetMapping("/embeddings")
    public Object embed(@RequestParam String text) {
        return embeddings.embed(text);
    }

    @GetMapping("/similarity")
    public Object similar(@RequestParam String text) {
        return embeddings.similar(text);
    }
}

File: src/main/kotlin/il/tutorials/similaritysearch/EmbeddingsController.kt
package il.tutorials.similaritysearch

import org.springframework.web.bind.annotation.*

@RestController
class EmbeddingsController(val embeddings: EmbeddingsService) {

    @GetMapping("/embeddings")
    fun embed(@RequestParam text: String) = embeddings.embed(text)

    @GetMapping("/similarity")
    fun similar(@RequestParam text: String) = embeddings.similar(text)
}